lib/internal/repl/utils.js

```diff
@@ -739,11 +739,25 @@ function setupReverseSearch(repl) {
   return { reverseSearch };
 }
 
+/**
+ * Checks if some provided code represents an object literal.
+ * This is helpful to prevent confusing repl code evaluations where
+ * strings such as `{ a : 1 }` would get interpreted as block statements
+ * rather than object literals.
+ * @param {string} code the code to check
+ * @returns {boolean} true if the code represents an object literal, false otherwise
+ */
+function isObjectLiteral(code) {
+  return RegExpPrototypeExec(/^\s*{/, code) !== null &&
```

Comment:
Can we move the regexes outside of the function to improve performance?

Reply:
Are you suggesting not to have the function at all?

If that's the case I can simply revert https://github.com/nodejs/node/pull/57498/commits/146f2682dabd382d9d3663786789f472151d639c

I've added the function so that we can have a single place where this code lives instead of repeating the regexes in two places and having comments saying that the two need to be kept in sync (meaning that we gain in maintainability but lose a bit in performance)
 
Since this is repl I also assume that performance are not a huge concern, especially for something like this run during the code evaluation

Anyways I'm happy to make the change if you believe that not having this function would be better :slightly_smiling_face:

Reply:
I mean declaring `/^\s*{/` and the other regex outside of the function so it doesn't get created multiple times.

Reply:
ahhhh, ok sorry I completely misunderstood the suggestion :sweat_smile::+1:

Reply:
done :slightly_smiling_face: